aws_lambda_powertools/utilities/idempotency/idempotency.py

```diff
@@ -66,7 +69,16 @@ def idempotent(
         >>>     return {"StatusCode": 200}
     """
 
-    if os.getenv(constants.IDEMPOTENCY_DISABLED_ENV):
+    # Skip idempotency controls when POWERTOOLS_IDEMPOTENCY_DISABLED has a truthy value
+    # Raises a warning if not running in development mode
+    if strtobool(os.getenv(constants.IDEMPOTENCY_DISABLED_ENV, "false")):
+        if not powertools_dev_is_set():
```

Comment:
Think is necessary? It's another if branching here to maintain and higher memory

I'd personally warn either way. If they're explicitly in dev, other warnings will appear, and they can also suppress them.

Reply:
To agree. I'm sure we have situations like this in other parts of the codebase and I will review them.